src/shogun/base/SGObject.h

```diff
@@ -1207,6 +1204,20 @@ class CSGObject
 		int64_t m_next_subscription_index;
 	};
 
+template <class T>
+T* make_clone(T* orig, ParameterProperties pp = ParameterProperties::ALL)
+{
+	REQUIRE(orig, "No object provided.\n");
+	return orig->clone(pp)->template as<T>();
```

Comment:
I guess you could actually just have a static_cast here no? you know it must be of type T so it shouldn't require a runtime check?

Reply:
yes